src/Handler/TagHandler.php

```diff
@@ -117,9 +119,14 @@ public function addTags(array $tags)
      */
     public function invalidateTags(array $tags)
     {
-        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $this->escapeTags($tags))));
-        $headers = [$this->tagsHeader => $tagExpression];
-        $this->invalidator->invalidate($headers);
+        if (empty($tags)) {
+            return $this;
+        }
+
+        $this->invalidator->invalidateTags(
+            $this->escapeTags($tags),
```

Comment:
ah, monday morning... now i realize that TagHandler is also about adding
tags to responses. i guess we want to keep that part. do you plan to use
a header for tagging responses with HttpCache as well?

either way we would want to keep that part. which raises the question
where the escaping should be. maybe still in the TagHandler but public
so that caching clients can use them? or should we keep invalidateTags
on TagHandler in the end and document on CacheInvalidator that the end
user is not allowed to call invalidateTags there, mark the public method
as `@private` in phpdoc?